Update Logging Properties Documentation (#21802)
prestodb/presto: presto-docs/src/main/sphinx/admin/properties.rst

    The name of the catalog to be used for Common Table Expressions (CTE) and which provides custom partitioning for Common Table Expression (CTE) materialization.
    This setting specifies which catalog should be used for CTE materialization and for determining how to partition the materialization of CTEs in queries.
    This can also be specified on a per-query basis using the ``cte_partitioning_provider_catalog`` session property.

Logging Properties
------------------

``log.path``
^^^^^^^^^^^^

    * **Type:** ``string``
    * **Default value:** ``var/log/server.log``

    The path to the log file used by Presto. The path is relative to the data
    directory, configured by the launcher script.

``log.max-history``
^^^^^^^^^^^^^^^^^^^

    * **Type:** ``integer``
    * **Default value:** ``30``

>     The maximum number of general application log files to use, before log

Reviewer: The same explanation holds true for this as well

    rotation replaces old content.

``log.max-size``
^^^^^^^^^^^^^^^^

    * **Type:** ``data size``
    * **Default value:** ``100MB``

    The maximum file size for the general application log file.

``http-server.log.enabled``
^^^^^^^^^^^^^^^^^^^^^^^^^^^

    * **Type:** ``boolean``
    * **Default value:** ``true``

    Flag to enable or disable logging for the HTTP server.

``http-server.log.compression.enabled``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

    * **Type:** ``boolean``
    * **Default value:** ``true``

    Flag to enable or disable compression of the log files of the HTTP server.

``http-server.log.path``
^^^^^^^^^^^^^^^^^^^^^^^^

    * **Type:** ``string``
    * **Default value:** ``var/log/http-request.log``

    The path to the log file used by the HTTP server. The path is relative to
    the data directory, configured by the launcher script as detailed in
    :ref:`running_presto`.

``http-server.log.max-history``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

    * **Type:** ``integer``
    * **Default value:** ``15``

>     The maximum number of log files for the HTTP server to use, before

Reviewer: `http-server.log.max-history` property is a little more complex than it is named. I have played around with this in the past and it doesn't directly correlate to the number of files.

So, Max History controls the number of archive log periods to keep. 
Log Period is determined from the FileNamePattern. In our case, it's a daily rollover (https://github.com/prestodb/airlift/blob/a5f238a6ba3426264fad89422fabaf5ef3ae0f7d/http-server/src/main/java/com/facebook/airlift/http/server/DelimitedRequestLog.java#L94).
So, 1 log period in our case will be 1 day. If we set `http-server.log.max-history` to 5, that would mean it will keep logs for 5 log periods (i.e. days). 

More information on TimeBasedRollingPolicy(which airlift is using) from logback: https://logback.qos.ch/manual/appenders.html#TimeBasedRollingPolicy

    The name of the catalog to be used for Common Table Expressions (CTE) and which provides custom partitioning for Common Table Expression (CTE) materialization.
    This setting specifies which catalog should be used for CTE materialization and for determining how to partition the materialization of CTEs in queries.
    This can also be specified on a per-query basis using the ``cte_partitioning_provider_catalog`` session property.

Logging Properties
------------------

> ``log.path``

Reviewer: @Akanksha-kedia Have we tested this config property? I tried to set this config but I am getting the below error

```
1) Error: Configuration property 'log.output-file' (=/Users/imjalpreet/Documents/PrestoRuntime/data/var/log/server.log) conflicts with property 'log.path' (=~/Downloads/server.log)

com.google.inject.ConfigurationException: Guice configuration errors:

1) Error: Configuration property 'log.output-file' (=/Users/imjalpreet/Documents/PrestoRuntime/data/var/log/server.log) conflicts with property 'log.path' (=~/Downloads/server.log)
```

^^^^^^^^^^^^

    * **Type:** ``string``
    * **Default value:** ``var/log/server.log``

    The path to the log file used by Presto. The path is relative to the data
    directory, configured by the launcher script.

``log.max-history``
^^^^^^^^^^^^^^^^^^^

    * **Type:** ``integer``
    * **Default value:** ``30``

>     The `log.max-history` property controls the number of archived log periods that the application retains.

Reviewer: Let me try and verify once that the same logic(as http-server) also applies to this config. If yes, then we can make the same change here as well that I have mentioned for http-server

>     The `log.max-history` property controls the number of archived log periods that the application retains.
>     In Presto, one log period corresponds to one day. For instance, if set to 30, the system will keep logs for the
>     past 30 days.

Reviewer: I have verified it has the same logic for this property as well.

minor nit: 
```suggestion
    The ``log.max-history`` property controls the number of archive log periods that the application retains.
    In Presto, one log period corresponds to one day. For instance, if ``log.max-history`` is set to 30, the system will keep logs for the
    past 30 days.
```

``log.max-size``
^^^^^^^^^^^^^^^^

    * **Type:** ``data size``
    * **Default value:** ``100MB``

    The maximum file size for the general application log file.

``http-server.log.enabled``
^^^^^^^^^^^^^^^^^^^^^^^^^^^

    * **Type:** ``boolean``
    * **Default value:** ``true``

    Flag to enable or disable logging for the HTTP server.

``http-server.log.compression.enabled``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

    * **Type:** ``boolean``
    * **Default value:** ``true``

    Flag to enable or disable compression of the log files of the HTTP server.

``http-server.log.path``
^^^^^^^^^^^^^^^^^^^^^^^^

    * **Type:** ``string``
    * **Default value:** ``var/log/http-request.log``

    The path to the log file used by the HTTP server. The path is relative to
    the data directory, configured by the launcher script as detailed in
    :ref:`running_presto`.

``http-server.log.max-history``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

    * **Type:** ``integer``
    * **Default value:** ``15``

>     The `http-server.log.max-history` property controls the number of archive log periods that the HTTP server retains.
>     In Presto, one log period corresponds to one day. For instance, if `http-server.log.max-history` is set to 15, the
>     system will keep logs for the past 15 days.

Reviewer: Please use double backticks instead of single
```suggestion
    The ``http-server.log.max-history`` property controls the number of archive log periods that the HTTP server retains.
    In Presto, one log period corresponds to one day. For instance, if ``http-server.log.max-history`` is set to 15, the
    system will keep logs for the past 15 days.
```

>     The `http-server.log.max-history` property controls the number of archived log periods that the HTTP server retains.
>     The 'log period' is determined by the FileNamePattern, configured to a daily rollover. Thus, one log period
>     equates to one day. For instance, if set to 15, the system will maintain logs for the past 15 days.

Reviewer: nit: I think since FileNamePattern is transparent to the presto users and cannot be controlled by users, we should try not to mention it in the documentation. 

```suggestion
    The `http-server.log.max-history` property controls the number of archive log periods that the HTTP server retains. In Presto, one log period corresponds to one day.
    For instance, if `http-server.log.max-history` is set to 15, the system will keep logs for the past 15 days.
```
